src/output/tree.rs

```diff
@@ -611,22 +616,37 @@ impl<'module> MonoidalPrinter<'module> {
         frags.finalize()
     }
 
+    fn is_indirect_format(&self, format: &Format) -> bool {
+        match format {
+            Format::ItemVar(..)
+            | Format::Dynamic(..)
+            | Format::Apply(..) => true,
+            _ => false,
+        }
+    }
+
+
     fn compile_field_value_continue(
         &mut self,
         scope: &Scope<'_>,
         label: impl fmt::Display,
         value: &Value,
         format: Option<&Format>,
+        format_needed: bool
     ) -> Fragment {
         let mut frags = FragmentBuilder::new();
         frags.push(self.compile_gutter());
         frags.push(Fragment::cat(
             Fragment::Symbol(Symbol::Junction),
             Fragment::String(format!("{label}").into()),
         ));
+
         if let Some(format) = format {
-            frags.push(Fragment::String(" <- ".into()));
-            frags.push(self.compile_format(format, Precedence::FORMAT_COMPOUND));
+            let tmp = self.compile_field_value(scope, value, Some(format));
+            if (self.is_indirect_format(format) && !tmp.is_single_line(true)) || format_needed || self.flags.show_redundant_formats {
+                frags.push(Fragment::String(" <- ".into()));
+                frags.push(self.compile_format(format, Precedence::FORMAT_COMPOUND));
+            }
         }
         self.gutter.push(Column::Branch);
         frags.push(self.compile_field_value(scope, value, format));
```

Comment:
Might be able to refactor this to avoid calling `compile_field_value` twice with the same arguments.

Reply:
Working on it right now. Luckily we are already monoidal so the fragment generation can happen out-of-order very easily.

Reply:
Done!